remote.go

```diff
@@ -539,6 +556,26 @@ func (c *RemoteCollection) Create(name string, url string) (*Remote, error) {
 	return remote, nil
 }
 
+func (c *RemoteCollection) CreateOptions(url string, option *RemoteCreateOptions) (*Remote, error) {
+	remote := &Remote{repo: c.repo}
+
+	curl := C.CString(url)
+	defer C.free(unsafe.Pointer(curl))
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	opts :=	remoteCreateOptionsToC(option)
+	defer freeRemoteCreateOptions(opts)
+	ret := C.git_remote_create_with_opts(&remote.ptr, curl, opts)
```

Comment:
```suggestion
	ret := C.git_remote_create_with_opts(&remote.ptr, curl, opts)
	runtime.KeepAlive(c.repo)
```

otherwise, the go runtime is completely free to free the repository under us while the Cgo call is being made.